File: sorting/insertion-sort/insertionSort.js
const insertionSoftAsc = data => {
    const sortedData = [...data];

    for (let i = 0; i < sortedData.length; i++) {
        let currentPosition = i;

        while (currentPosition > 0 && sortedData[currentPosition - 1] > sortedData[currentPosition]) {
            [sortedData[currentPosition - 1], sortedData[currentPosition]] = [
                sortedData[currentPosition],
                sortedData[currentPosition - 1],
            ];
            currentPosition--;
        }
    }

    return sortedData;
};

const d = [50, 5, 7, 98, 4, 2, 5, 7, 9];
const resultAsc = insertionSoftAsc(d);
console.log(resultAsc);

File: sorting/insertion-sort/insertionSort.go
package main

import "fmt"

func insertionSoftAsc(data []int) []int {
	sortedData := make([]int, len(data))
	copy(sortedData, data)

	for i := range sortedData {
		currentPosition := i

		for currentPosition > 0 && sortedData[currentPosition-1] > sortedData[currentPosition] {
			sortedData[currentPosition-1], sortedData[currentPosition] = sortedData[currentPosition], sortedData[currentPosition-1]
			currentPosition--
		}
	}

	return sortedData
}

func main() {
	d := []int{50, 5, 7, 98, 4, 2, 5, 7, 9}
	resultAsc := insertionSoftAsc(d)
	fmt.Println(resultAsc)
}
